Invalid `pointless-string-statement` for docstrings for Python 3.12 type-aliases (PEP 695)

### Bug description

```python
Adding a multiline docstring to a Python 3.12 type-alias (PEP 695) causing pylint to faild with a `pointless-string-statement`.

Example code:

type _GeneratorCallable = Callable[..., list[str]]
"""
Multiline
docstring
for a new type-alias
introduced in Python3.12
"""


Adding a `:` before the string also doesn't help:
```python
type _GeneratorCallable = Callable[..., list[str]]
""":
Multiline
docstring with :
for a new type-alias
introduced in Python3.12
"""
```
```


### Configuration

_No response_

### Command used

```shell
pylint src
```


### Pylint output

```python
************* Module src.generator.generator
src/generator/generator.py:8:0: W0105: String statement has no effect (pointless-string-statement)
```


### Expected behavior

Pylint should not raise a warning here

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.12.0 (main, Nov  2 2023, 13:03:57) [Clang 15.0.0 (clang-1500.0.40.1)]
```


### OS / Environment

macOS Sonoma, iTerm2

### Additional dependencies

_No response_

--- a/doc/data/messages/p/pointless-string-statement/related.rst
+++ b/doc/data/messages/p/pointless-string-statement/related.rst
@@ -1,0 +1,1 @@
+- `Discussion thread re: docstrings on assignments <https://discuss.python.org/t/docstrings-for-new-type-aliases-as-defined-in-pep-695/39816>`_

--- a/doc/whatsnew/fragments/9268.false_positive
+++ b/doc/whatsnew/fragments/9268.false_positive
@@ -1,0 +1,4 @@
+Fixed ``pointless-string-statement`` false positive for docstrings
+on Python 3.12 type aliases.
+
+Closes #9268

--- a/pylint/checkers/base/basic_checker.py
+++ b/pylint/checkers/base/basic_checker.py
@@ -446,7 +446,9 @@ class BasicChecker(_BasicChecker):
                     if (
                         sibling is not None
                         and sibling.scope() is scope
-                        and isinstance(sibling, (nodes.Assign, nodes.AnnAssign))
+                        and isinstance(
+                            sibling, (nodes.Assign, nodes.AnnAssign, nodes.TypeAlias)
+                        )
                     ):
                         return
             self.add_message("pointless-string-statement", node=node)

--- a/tests/functional/s/statement_without_effect_py312.py
+++ b/tests/functional/s/statement_without_effect_py312.py
@@ -1,0 +1,7 @@
+"""Move this into statement_without_effect.py when python 3.12 is minimum."""
+
+type MyTuple = tuple[str, str]
+"""
+Multiline docstring
+for Python3.12 type alias (PEP 695)
+"""

--- a/tests/functional/s/statement_without_effect_py312.rc
+++ b/tests/functional/s/statement_without_effect_py312.rc
@@ -1,0 +1,2 @@
+[testoptions]
+min_pyver=3.12

